scenes/ui_elements/dialogue/components/balloon.gd

```diff
@@ -211,4 +222,40 @@ func _on_balloon_gui_input(event: InputEvent) -> void:
 func _on_responses_menu_response_selected(response: DialogueResponse) -> void:
 	next(response.next_id)
 
+
 #endregion
+
+
+func _on_spoke(_letter: String, _index: int, _speed: float) -> void:
+	if _has_played_sfx:
+		return
+
+	_has_played_sfx = true
+
+	if talk_sound_player.stream != SFX_MAP["pencil"]:
+		talk_sound_player.stream = SFX_MAP["pencil"]
+
+	talk_sound_player.pitch_scale = 1.0
+	talk_sound_player.volume_db = 2
+	talk_sound_player.play()
+	talk_sound_player.seek(_paused_position)
+
+
+func _on_dialogue_finished_typing() -> void:
+	if talk_sound_player.playing:
+		_paused_position = talk_sound_player.get_playback_position()
+		talk_sound_player.stop()
+	else:
+		_paused_position = 0.0
```

Comment:
I think the `else` will never happen, because you have a callback `_on_talk_sound_finished` that automatically plays the SFX when finished, basically creating a custom loop.

This brings me a question, can the stream be set to loop and then you just pause/resume instead of using a separate variable `_paused_position`?